### PR TITLE
chore(dec-config): add tests to access scope updater

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Improve the test coverage of the access scope updater for declarative config.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- see CI.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
* **#10421** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10421?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10420** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10420?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10419** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10419?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10418** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10418?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dhaus67 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: central/declarativeconfig/health/datastore/datastore.go [hunk 1]
```diff
@@ -24,3 +27,8 @@ func New(storage store.Store) DataStore {
 		store: storage,
 	}
 }
+
+// GetTestPostgresDataStore provides a datastore connected to postgres for testing purposes.
+func GetTestPostgresDataStore(_ *testing.T, pool postgres.DB) DataStore {
```
Comment (janisz) [inline central/declarativeconfig/health/datastore/datastore.go:32]: Importing `testing` may have unexpected consequences 
https://github.com/stackrox/stackrox/blob/ec3a4a07d00c4edc44e4334fa49a8bdd73064503/tools/roxvet/common/file_filters.go#L28-L30

File: central/declarativeconfig/updater/access_scope_updater_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+				s.True(exists)
+			}
+		})
+	}
+}
+
+func (s *updaterTestSuite) TestDelete_Successful() {
+	scopes := []*storage.SimpleAccessScope{
+		{
+			Id:          "61a68f2a-2599-5a9f-a98a-8fc83e2c06cf",
```
Comment (janisz) [inline central/declarativeconfig/updater/access_scope_updater_test.go:94]: How about ids that are easier to remember
```suggestion
			Id:          "11111-....",
```

**Comment by dhaus67** [comment]
@rukletsov will do the rename of the `Delete` function as the last PR in the series to avoid too much rebasing.

File: pkg/uuid/uuid.go [hunk 3]
```diff
@@ -152,3 +153,13 @@ func NewV5(ns UUID, name string) UUID {
...
+
+// NewTestUUID returns a UUID for testing purposes with the given number.
+// Example: 11111111-1111-1111-1111-111111111111.
+func NewTestUUID(d int) UUID {
+	return FromStringOrNil(fmt.Sprintf(
+		"%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d-"+
+			"%[1]d%[1]d%[1]d%[1]d-"+
+			"%[1]d%[1]d%[1]d%[1]d-"+
+			"%[1]d%[1]d%[1]d%[1]d-%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d%[1]d", d))
+}
```
Comment (janisz) [inline pkg/uuid/uuid.go:165]:
```suggestion
	s := fmt.Sprintf("%d", d)
	return FromStringOrNil(
		strings.Repeat(s, 8) +
			strings.Repeat(s, 4) +
			strings.Repeat(s, 4) +
			strings.Repeat(s, 4) +
			strings.Repeat(s, 12))
}
```

File: pkg/uuid/uuid.go [hunk 4]
```diff
@@ -152,3 +153,13 @@ func NewV5(ns UUID, name string) UUID {
 func NewDummy() UUID {
 	return FromStringOrNil("aaaaaaaa-bbbb-4011-0000-111111111111")
 }
+
+// NewTestUUID returns a UUID for testing purposes with the given number.
```
Comment (janisz) [inline pkg/uuid/uuid.go:157]:
```suggestion
// NewTestUUID returns a UUID for testing purposes with the given number.
// If number is negative or greater then 9 then zeroes are returned
```

**Review by janisz — APPROVED** [review]

File: pkg/uuid/uuid.go (hunk 4 above)
Comment (janisz) [reply pkg/uuid/uuid.go:157]: You may add test case for that as well

**Comment by dhaus67** [comment]
### Merge activity

* **Apr 7, 1:19 AM EDT**: @dhaus67 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/10418).

**Pull request merged**